cartographer/mapping_2d/pose_graph/optimization_problem.cc

```diff
@@ -52,6 +57,63 @@ transform::Rigid2d ToPose(const std::array<double, 3>& values) {
   return transform::Rigid2d({values[0], values[1]}, values[2]);
 }
 
+// Selects a trajectory node closest in time to the landmark observation and
+// applies a relative transform from it.
```

Comment:
Why not interpolate here as well? Sure, it's just the starting solution so it doesn't matter as much, but since you already went to the trouble of determining the closest node, I wonder why not interpolate.

Reply:
Oh, it will be interpolated for sure. It was just a drafty solution to make a demo.